### DISCUSSION TITLE
ssh private key not being used

### DISCUSSION THREAD

**Discussion by huangjy-pku** [community/community#28961] [Other Feature Feedback, Questions, & Ideas] [answered]
I have already generated a new ssh key and added it to my github account. And I have passed the test command `ssh -T git@github.com`. However, when I went to clone a repo in my github account, I met the following problem:

> git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Next I carefully checked the documentation [https://docs.github.com/en/authentication/troubleshooting-ssh/error-permission-denied-publickey](url).

I found that it worked well in [https://docs.github.com/en/authentication/troubleshooting-ssh/error-permission-denied-publickey#make-sure-you-have-a-key-that-is-being-used](url), which means when I run `ssh-add -l -E sha256` it returns the proper key that matches the saved keys associated with my github account.

But I failed the next part in [https://docs.github.com/en/authentication/troubleshooting-ssh/error-permission-denied-publickey#getting-more-details](url) (`ssh -vT git@github.com`), which shows the identity files at -1 type and indicates my private key is not being used.

Til now, I am quite confused why this could happen. By the way, I wonder if it stems from my Win11 OS, since I easily finished this procedure on my another Ubuntu machine before :<

**Comment by dongho-jung** [comment, marked as answer]
How about `GIT_SSH_COMMAND='ssh -i ./your-key-file-location -o IdentitiesOnly=yes' git clone ...`? does it make any change? and also checkout your ssh-agent is pointing your ket file properly

**Reply by huangjy-pku** [reply]
Thanks, but it seems the `GIT_SSH_COMMAND` not compatible with Win terminal. So I tried it with `git bash`, and it worked! So, does this mean the ssh-agent is not informed of the path of private key? I do place the keys in the default location `~/.ssh/`.

**Reply by dongho-jung** [reply]
which terminal you use now? cmd or PowerShell? I did what you want to do before in PowerShell so hopefully and technically your requirements aren't impossible. I think [this](https://stackoverflow.com/questions/48843643/using-git-with-powershell-and-ssh-key-with-passphrase) might help you

In summary,
1. check out your `ssh-agent` in Windows Services. find "OpenSSH Authentication Agent" in services.msc and make it running status
2. `git config --global core.sshCommand C:/Windows/System32/OpenSSH/ssh.exe`
3. add below to `C:\Users\%YOUR_USERNAME%\.ssh\config`
```
Host *
    AddKeysToAgent yes
    IdentitiesOnly yes
```
4. `ssh-add $HOME/.ssh/your_file_name`
5. restart your PowerShell

**Reply by huangjy-pku** [reply]
Thanks a lot! I follow your instructions and manage to clone repos in my powershell. And I make some further trials to figure out the underlying critical factor. I find that even without `AddKeysToAgent` and `IdentitiesOnly` options, simply re-configuring the ssh client `git config --global core.sshCommand C:/Windows/System32/OpenSSH/ssh.exe` is enough to fix my problem. So, maybe it is the ssh running option that truly matters :)
